Sources/ManageSettings.php

```diff
@@ -253,6 +253,27 @@ function ModifyBasicSettings($return_config = false)
 	{
 		checkSession();
 
+		// Make sure the country codes are valid.
+		if (!empty($_POST['timezone_priority_countries']))
+		{
+			$country_codes = explode(',', $_POST['timezone_priority_countries']);
+			foreach ($country_codes as $key => $country_code)
+			{
+				$country_code = substr(trim($country_code), 0, 2);
+				$country_tzids = @timezone_identifiers_list(DateTimeZone::PER_COUNTRY, $country_code);
+
+				if (empty($country_tzids))
+					unset($country_codes[$key]);
+				else
+					$country_codes[$key] = $country_code;
+			}
+
+			if (empty($country_codes))
+				unset($_POST['timezone_priority_countries']);
+			else
+				$_POST['timezone_priority_countries'] = implode(',', $country_codes);
+		}
+
```

Comment:
Can this code be moved to `Subs-Timezones.php`under a new function name such as `validateTimezones()`?

Reply:
Good idea.

Reply:
This is a quote from a member on my site. He says this:


Thanks, I've had a look at his fix. It will cause bugs you can't  take the first 2 letters of a 3 letter code. Some countries 2 and 3 letter codes change.
E.G Bangladesh's 2 letter is BD its 3 letter is BGD if you substitute BGD to BG that is Bulgaria.

They might want to look at the 1st and 3rd letter they might match better?

Reply:
Sorry the fix did not work...see quote above.

Reply:
The member on your site misunderstands the purpose of this part of the code. It is not meant to correct three-character codes to two-character codes. It is just removing white space and any extra characters that might have snuck into the input for whatever reason.

In your original "USA" example, truncating to two characters coincidentally happened to produce the correct country code, but that was just a happy accident. Generally speaking, if the truncation happens to produce a valid but unexpected result, it will be the responsibility of the admin to find and enter the correct two-character code for the intended country.

Reply:
...Actually, now that I think about it, it's probably cleaner to just fail immediately on overlong strings, rather than truncating them. If the admin accidentally enters an extra character, we'll just reject the whole string.

Reply:
I like that approach to play hard ball. Anything to reduce unexpected behavior is a good thing imo.